client/post-editor/editor-term-selector/index.jsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import map from 'lodash/map';
+import cloneDeep from 'lodash/cloneDeep';
+
+/**
+ * Internal dependencies
+ */
+import TermTreeSelector from 'my-sites/term-tree-selector';
+import PostActions from 'lib/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class EditorTermSelector extends Component {
+	static propTypes = {
+		siteId: React.PropTypes.number,
+		postTerms: React.PropTypes.object,
+		taxonomyName: React.PropTypes.string
+	};
+
+	constructor( props ) {
+		super( props );
+		this.boundOnTermsChange = this.onTermsChange.bind( this );
+	}
+
+	onTermsChange( selectedTerm ) {
+		const { postTerms, taxonomyName } = this.props;
+		const terms = cloneDeep( postTerms ) || {};
+		const taxonomyTerms = terms[ taxonomyName ];
+
+		if ( taxonomyTerms[ selectedTerm.name ] ) {
```

Comment:
I think it'd be sufficient to add `|| {}` fallback for `const taxonomyTerms`, and I was going to commit this, but encountered another issue where trying to remove a term from a saved CPT would not persist the removal on save.